apps/nextjs/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,17 +1,37 @@
 import { NextRequest } from "next/server";
 
 import { createHandlers } from "@homarr/auth";
+import type { SupportedAuthProvider } from "@homarr/definitions";
 import { logger } from "@homarr/log";
 
 export const GET = async (req: NextRequest) => {
-  return await createHandlers(isCredentialsRequest(req)).handlers.GET(reqWithTrustedOrigin(req));
+  return await createHandlers(extractProvider(req)).handlers.GET(reqWithTrustedOrigin(req));
 };
 export const POST = async (req: NextRequest) => {
-  return await createHandlers(isCredentialsRequest(req)).handlers.POST(reqWithTrustedOrigin(req));
+  return await createHandlers(extractProvider(req)).handlers.POST(reqWithTrustedOrigin(req));
 };
 
-const isCredentialsRequest = (req: NextRequest) => {
-  return req.url.includes("credentials") && req.method === "POST";
+/**
+ * This method extracts the used provider from the url and allows us to override the getUserByEmail method in the adapter.
+ * @param req request containing the url
+ * @returns the provider or "unknown" if the provider could not be extracted
+ */
+const extractProvider = (req: NextRequest): SupportedAuthProvider | "unknown" => {
+  const url = new URL(req.url);
+
+  if (url.pathname.includes("oidc")) {
+    return "oidc";
+  }
+
+  if (url.pathname.includes("credentials")) {
+    return "credentials";
+  }
+
+  if (url.pathname.includes("ldap")) {
+    return "ldap";
+  }
+
+  return "unknown";
```

Comment:
log this?

Reply:
No, because otherwise it will always logs when you request your session. It's only necessary for the getUserByEmail function